packages/flutter_tools/bin/macos_build_flutter_assets.sh

```diff
@@ -40,6 +40,37 @@ if [[ -n "$TRACK_WIDGET_CREATION" ]]; then
   track_widget_creation_flag="--track-widget-creation"
 fi
 
+# Copy the framework and handle local engine builds.
+derived_dir="${SOURCE_ROOT}/../build/macos/Build/Products/flutter_framework"
+framework_path="${FLUTTER_ROOT}/bin/cache/artifacts/engine/darwin-x64"
+flutter_framework="${framework_path}/FlutterMacOS.framework"
+
+if [[ -n "$FLUTTER_ENGINE" ]]; then
+  flutter_engine_flag="--local-engine-src-path=${FLUTTER_ENGINE}"
+fi
+
+if [[ -n "$LOCAL_ENGINE" ]]; then
+  if [[ $(echo "$LOCAL_ENGINE" | tr "[:upper:]" "[:lower:]") != *"$build_mode"* ]]; then
+    EchoError "========================================================================"
+    EchoError "ERROR: Requested build with Flutter local engine at '${LOCAL_ENGINE}'"
+    EchoError "This engine is not compatible with FLUTTER_BUILD_MODE: '${build_mode}'."
+    EchoError "You can fix this by updating the LOCAL_ENGINE environment variable, or"
+    EchoError "by running:"
+    EchoError "  flutter build macos --local-engine=host_${build_mode}"
+    EchoError "or"
+    EchoError "  flutter build macos --local-engine=host_${build_mode}_unopt"
+    EchoError "========================================================================"
+    exit -1
+  fi
+  local_engine_flag="--local-engine=${LOCAL_ENGINE}"
+  flutter_framework="${FLUTTER_ENGINE}/out/${LOCAL_ENGINE}/FlutterMacOS.framework"
+fi
+
+RunCommand mkdir -p -- "$derived_dir"
+RunCommand rm -rf -- "${derived_dir}/FlutterMacOS.framework"
+RunCommand cp -r -- "${flutter_framework}" "${derived_dir}"
```

Comment:
s/-r/-R/ to make sure everything is preserved correctly.

Reply:
Done

Reply:
Actually, I just noticed that this is running every time. Is that the case on iOS as well, or is there some kind of stamp file?

If we're going to copy every time we should probably use `-Rp` as we do in the current FDE example, because if we don't preserve the timestamps I would expect that this would cause an unnecessary relink for no-op builds.

Reply:
This is the case on iOS too. There was a fingerprint check to attempt skipping the entire build, which I just disabled because it ended up not being correct (and we don't have the tests to keep it correct)